runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheClientConsumerFactory.java

```diff
@@ -154,15 +154,16 @@ public MessageConsumer newStream(
         {
             final long resolvedId = resolved.id;
 
-            KafkaCacheClientConsumerFanout fanout = clientConsumerFansByGroupId.get(groupId);
+            String consumer = String.format("%s-%s-%s-%d", groupId, topic, consumerId, resolvedId);
+            KafkaCacheClientConsumerFanout fanout = clientConsumerFansByConsumer.get(consumer);
```

Comment:
Consider renaming `...Fanout` and `fanout` to `...Fan` and `fan` since the fan behavior is both fan-in and fan-out.